### PR TITLE
Declare variables global/local

### PR DESCRIPTION
**Pull request by alyssais**
This caused warnings with `setopt warn_create_global`

### CODE REVIEW

**Comment by olivierverdier** [comment]
Nice! Thank you!

**Comment by alyssais** [comment]
@olivierverdier any chance we can get this merged?

File: zshrc.sh [hunk 1]
```diff
@@ -43,28 +43,30 @@ function chpwd_update_git_vars() {
...
-     __CURRENT_GIT_STATUS=("${(@s: :)_GIT_STATUS}")
-	GIT_BRANCH=$__CURRENT_GIT_STATUS[1]
-	GIT_AHEAD=$__CURRENT_GIT_STATUS[2]
-	GIT_BEHIND=$__CURRENT_GIT_STATUS[3]
-	GIT_STAGED=$__CURRENT_GIT_STATUS[4]
-	GIT_CONFLICTS=$__CURRENT_GIT_STATUS[5]
-	GIT_CHANGED=$__CURRENT_GIT_STATUS[6]
-	GIT_UNTRACKED=$__CURRENT_GIT_STATUS[7]
+    
+    typeset -ga __CURRENT_GIT_STATUS=("${(@s: :)_GIT_STATUS}")
```
Comment (olivierverdier) [inline zshrc.sh:55]: I get an error on this line: `no matches found: __CURRENT_GIT_STATUS=()`. I don't know enough zsh to understand that...

Comment (alyssais) [reply zshrc.sh:55]: @olivierverdier Python or Haskell?

Comment (alyssais) [reply zshrc.sh:55]: My guess is this is happening because `_GIT_STATUS` is empty, but I can't see why this change would have affected that…

**Comment by alyssais** [comment]
As is probably obvious, I don’t plan on working on this any more :(

**Pull request closed without merging**